### DISCUSSION TITLE
Building from source on Windows

### DISCUSSION THREAD

**Discussion by ivancopc** [ImageMagick/ImageMagick#4705] [General]
I need support for tiff images with jpeg compression. To do this, I built the tiff library with support for working with jpeg compression. How can I build ImageMagick using my version of the tiff library to support jpeg compressed tiff images? I am using windows 10 operating system and visual studio 2017 64 bit.

**Comment by snibgo** [comment]
The pre-built Windows binary can build TIFFs with JPEG compression, eg:
```
magick in.png -compress jpeg out.tiff
```

**Reply by ivancopc** [reply]
I am using ImageMagick as a static library in my project. Also, I use ImageMagick and LibTiff library at the same time in my project. If library LibTiff is different from the one used in ImageMagick, then I get linker errors. Therefore, it is important for me to build ImageMagick using the library LibTiff I built, but I do not know how to specify this when configuring.

**Comment by dlemstra** [comment]
More information on building with windows can be found here: https://github.com/ImageMagick/ImageMagick-Windows. This will help you with the project setup.
